File: raster/src/main/java/com/kjipo/raster/segment/SegmentImpl.java
package com.kjipo.raster.segment;

import javax.xml.bind.annotation.XmlRootElement;
import java.util.List;

@XmlRootElement
public class SegmentImpl implements Segment {
    private List<Pair> pairs;


    public SegmentImpl() {

    }


    public SegmentImpl(List<Pair> pairs) {
        if (pairs.isEmpty()) {
            throw new IllegalArgumentException("List of pairs cannot be empty");
        }
        this.pairs = pairs;
    }


    @Override
    public List<Pair> getPairs() {
        return pairs;
    }

    @Override
    public String toString() {
        return "SegmentImpl{" +
                "pairs=" + pairs +
                '}';
    }
}

File: encodings-multimodule/src/commonMain/kotlin/com/kjipo/representation/segment/SegmentImpl.kt
package com.kjipo.representation.segment


class SegmentImpl : Segment {
    override var pairs: List<Pair> = mutableListOf()
        private set

    constructor() {}

    constructor(pairs: List<Pair>) {
        this.pairs = pairs
    }

    override fun toString(): String {
        return "SegmentImpl{" +
                "pairs=" + pairs +
                '}'
    }
}